internal/k8s/controllers/node_controller_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"go.universe.tf/metallb/internal/config"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestNodeController(t *testing.T) {
+	var testNodeName = "testnode"
+	tests := []struct {
+		desc                 string
+		handlerRes           SyncState
+		expectReconcileFails bool
+	}{
+		{
+			desc:                 "handler returns SyncStateSuccess",
+			handlerRes:           SyncStateSuccess,
+			expectReconcileFails: false,
+		},
+		{
+			desc:                 "handler returns SyncStateError",
+			handlerRes:           SyncStateError,
+			expectReconcileFails: true,
+		},
+		{
+			desc:                 "handler returns SyncStateErrorNoRetry",
+			handlerRes:           SyncStateErrorNoRetry,
+			expectReconcileFails: false,
+		},
+		{
+			desc:                 "handler returns SyncStateReprocessAll",
+			handlerRes:           SyncStateReprocessAll,
+			expectReconcileFails: false,
+		},
+	}
+	for _, test := range tests {
+		fakeClient := newFakeClient(t, config.ClusterResources{})
+		err := fakeClient.Create(context.TODO(), &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testNodeName,
+				Namespace: testNamespace,
+			}})
+
+		mockHandler := func(log.Logger, *corev1.Node) SyncState {
+			return test.handlerRes
+		}
+
+		if err != nil {
+			t.Fatalf("test %s failed, got unexpected error: %v", test.desc, err)
```

Comment:
maybe worth mentioning is failed creating the fakeClient

Reply:
done